src/Qir/Execution/CommandLineCompiler/Program.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Help;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Tools;
+
+namespace CommandLineCompiler
+{
+    class Program
+    {
+        private static async Task<int> Main(string[] args)
+        {
+            var buildCommand = CreateBuildCommand();
+
+            var root = new RootCommand() { buildCommand };
+            SetSubCommandAsDefault(root, buildCommand);
+            root.Description = "Command-line tool for processing QIR DLL files.";
+            root.TreatUnmatchedTokensAsErrors = true;
+
+            Console.OutputEncoding = Encoding.UTF8;
+            return await new CommandLineBuilder(root)
+                .UseDefaults()
+                .UseHelpBuilder(context => new QsHelpBuilder(context.Console))
+                .Build()
+                .InvokeAsync(args);
+        }
+
+        /// <summary>
+        /// Creates the Build command for the command line compiler, which is used to
+        /// compile the executables from a given QIR DLL.
+        /// </summary>
+        /// <returns>The Build command.</returns>
+        private static Command CreateBuildCommand()
+        {
+            var buildCommand = new Command("build", "(default) Build the executables from a QIR DLL.")
+            {
+                Handler = CommandHandler.Create((BuildOptions settings) =>
+                {
+                    if (settings.LibraryDirectories.Length < 1)
+                    {
+                        throw new ArgumentException("The '--libraryDirectories' option requires at least one argument.");
+                    }
+                    return QirTools.BuildFromQSharpDll(settings.QsharpDll, settings.LibraryDirectories[0], settings.IncludeDirectory, settings.ExecutablesDirectory);
```

Comment:
It is very odd to me that the libraries directories takes an array of libraries and then ignores all but the first. 
Why not allow a list of directories for both the libraries and the includes? Alternatively, if it should be one, then the command line should reflect that.

Reply:
I believe the API calls will take multiple library directory paths in the near future, perhaps in this PR. @cesarzc?

Reply:
Yes, Scott is right. In the near future `BuildFromQSharpDll` will take a `IEnumerable` of `DirectoryInfo` for both libraries and include directories.

Reply:
@cesarzc Should I change the include directories parameter to take multiple arguments as well?

Reply:
I've updated the include directories to take multiple arguments as well.

Reply:
This is addressed now that the API also accepts arrays for the libraries and headers paths.